api/client.go

```diff
@@ -0,0 +1,943 @@
+/*
+Copyright 2020 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package api holds the implementation of the gRPC auth client
+package api
+
+import (
+	"compress/gzip"
+	"context"
+	"crypto/tls"
+	"io"
+	"net"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/golang/protobuf/ptypes/empty"
+	"github.com/gravitational/teleport"
+	"github.com/gravitational/teleport/api/auth"
+	"github.com/gravitational/teleport/lib/events"
+	"github.com/gravitational/teleport/lib/jwt"
+	"github.com/gravitational/teleport/lib/services"
+	"github.com/gravitational/teleport/lib/session"
+	"github.com/gravitational/trace"
+	"github.com/gravitational/trace/trail"
+	"golang.org/x/net/http2"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	ggzip "google.golang.org/grpc/encoding/gzip"
+	"google.golang.org/grpc/keepalive"
+)
+
+func init() {
+	if err := ggzip.SetLevel(gzip.BestSpeed); err != nil {
+		panic(err)
+	}
+}
+
+// Client is a gRPC Client that connects to a teleport auth server through TLS.
+type Client struct {
+	c    Config
+	grpc auth.AuthServiceClient
+	conn *grpc.ClientConn
```

Comment:
I know these are internal and don't require comments, but I would still add them here.